plugins/embed-optimizer/hooks.php

```diff
@@ -325,44 +325,13 @@ function embed_optimizer_lazy_load_scripts(): void {
  * @since 0.2.0
  */
 function embed_optimizer_get_lazy_load_script(): string {
-	return <<<JS
-		const lazyEmbedsScripts = document.querySelectorAll( 'script[type="application/vnd.embed-optimizer.javascript"]' );
-		const lazyEmbedScriptsByParents = new Map();
+	$script = file_get_contents( __DIR__ . '/embed-optimizer-lazy-load.js.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
```

Comment:
I don't think this file needs to be prefixed with `embed-optimizer-`. Just `lazy-load.js` is sufficient. Also, I see you've duplicated the `.js` at the end.

```suggestion
	$script = file_get_contents( __DIR__ . '/lazy-load.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
```